src/manage_nsfs/nc_lifecycle.js

```diff
@@ -210,23 +214,31 @@ async function process_rules(config_fs, bucket_json, object_sdk, should_notify)
     }
 }
 
-async function send_lifecycle_notifications(delete_res, bucket_json, object_sdk) {
+function create_notification_delete_object(delete_res, delete_obj_info) {
+    return {
+        ...delete_obj_info,
+        created_delete_marker: delete_res.created_delete_marker
+    };
+}
+
+async function send_lifecycle_notifications(delete_res, delete_candidates, bucket_json, object_sdk) {
     const writes = [];
-    for (const deleted_obj of delete_res) {
-        if (delete_res.err_code) continue;
+    for (let i = 0; i < delete_res.length; ++i) {
+        if (delete_res[i].err_code) continue;
         for (const notif of bucket_json.notifications) {
             if (notifications_util.check_notif_relevant(notif, {
                 op_name: 'lifecycle_delete',
-                s3_event_method: deleted_obj.delete_marker_created ? 'DeleteMarkerCreated' : 'Delete',
+                s3_event_method: delete_res[i].delete_marker_created ? 'DeleteMarkerCreated' : 'Delete',
             })) {
+                const deleted_obj = create_notification_delete_object(delete_res[i], delete_candidates[i]);
```

Comment:
are you sure that the delete_res[i] and delete_candidates[i] are the same object?

Reply:
they should match. it is what we do in s3_post_bucket_delete. the alternative is to modify the return object of of delete_multiple_objects to be like in containerized. for this PR I wanted to match as much as possible to s3_post_bucket_delete. as I don't know why this way was decided. currently it shouldn't matter as we can't have both delete marker and regular delete at the same time. but I would still validate this